boba/parser.py

```diff
@@ -387,9 +425,37 @@ def _warn_size(self):
                 print('Aborted.')
                 exit(0)
 
-    def main(self, verbose=True):
+    def main(self, decisions=None, verbose=True):
         self._warn_size()
-        self._code_gen()
+
+        if decisions is not None:
```

Comment:
perhaps move this chunk of code to a new function in the parser class (or a function in the decision parser class). We want to keep this `main` function nice and simple. Also, as we've discussed, placeholder variables are not allowed to have the same name as code blocks, so maybe we'll need some revisions related to this.